include/fmt/format-inl.h

```diff
@@ -155,7 +155,9 @@ FMT_FUNC void report_error(format_func func, int error_code,
   memory_buffer full_message;
   func(full_message, error_code, message);
   // Don't use fwrite_fully because the latter may throw.
-  (void)std::fwrite(full_message.data(), full_message.size(), 1, stderr);
+  if (std::fwrite(full_message.data(), full_message.size(), 1, stderr)) {
+      ;
+  }
   std::fputc('\n', stderr);
```

Comment:
I think we shouldn't be trying to write `\n` if an error occurs which will also solve the warning: 

```c++
  if (std::fwrite(full_message.data(), full_message.size(), 1, stderr) > 0)
    std::fputc('\n', stderr);
```